### DISCUSSION TITLE
generated adaptors/proxy: differentiate similarly named methods/properties

### DISCUSSION THREAD

**Discussion by 12-3-8-s9b9o9j9t** [Kistler-Group/sdbus-cpp#276] [General]
I saw that bluez (this bad boy again) has similarly named methods `RegisterApplication` and `UnregisterApplication`, with the same arguments, under the interfaces `org.bluez.GattManager1` and `org.bluez.Media1`.

When creating a proxy to this object with the convenient API, we can differentiate them thanks to `.onInterface()`, but how could we do it using only one generated proxy implementing both interfaces ?

**Comment by sangelovic** [comment]
@12-3-8-s9b9o9j9t In C++, you can simply use scope resolution operator as a qualified lookup to tell the compiler which variant of the method (from which base class) you want to be called. Like so:

```c++
MyBluezProxy proxy; // MyBluezProxy inherits from both generated interface classes org::bluez::GattManager1_proxy and org::bluez::Media1_proxy
proxy.org::bluez::GattManager1_proxy::RegisterApplication(); // RegisterApplication() on GattManager1_proxy interface will be called
proxy.org::bluez::Media1_proxy::RegisterApplication(); // RegisterApplication() on Media1_proxy interface will be called
```
